platform/core/src/utils/metadataProvider/fetchPaletteColorLookupTableData.js

```diff
@@ -13,12 +13,20 @@
  */
 export default function fetchPaletteColorLookupTableData(item, tag, descriptorTag) {
   const { PaletteColorLookupTableUID } = item;
-  const paletteData = item[tag];
+  let paletteData = item[tag];
   if (paletteData === undefined && PaletteColorLookupTableUID === undefined) {
     return;
   }
+  // Handle dicomfile loader
+  if (typeof paletteData === 'object' && paletteData[0] instanceof ArrayBuffer) {
+    // Convert ArrayBuffer to Uint8Array for processing
+    paletteData = {
+      InlineBinary: btoa(String.fromCharCode.apply(null, new Uint8Array(paletteData[0]))),
```

Comment:
What about new Uint8Array(paletteData[0])
This is doing a double conversion on a potentially 64k array - please update so it only does the conversion once.